### PR TITLE
abstracted table functionality to TableComponent.tsx and used this fo…

### PR DESCRIPTION
**Pull request by rabanspiegel**
…r SupplierTable and SupplierPartTable

### CODE REVIEW

File: src/components/supplier-detail/SupplierPartTable.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  const { tableData, isLoading } = useFetchParts();
+
+  return (
+    <>
+      {isLoading ? (
+        <Spinner />
+      ) : (
+        <TableComponent
+          columns={columns}
+          dataFetcher={() => Promise.resolve(tableData)}
```
Comment (bhausleitner) [inline src/components/supplier-detail/SupplierPartTable.tsx:62]: Passing down a promise makes the `TableComponent` more complex than it needs to be, I think we can just pass it the actual data that we receive from `api.part.partsBySupplier.useQuery();`

Comment (bhausleitner) [reply src/components/supplier-detail/SupplierPartTable.tsx:62]: You should be able to omit the the `useFetchParts` function replace the `SupplierPartTable` by something like this: 

```ts
export function SupplierTable() {
  const { data, isLoading } = api.supplier.getAllSuppliers.useQuery();
  return (
    <>
      {isLoading ? (
        <Spinner />
      ) : (
        <TableComponent
          columns={columns}
          data={data}
          filterPlaceholder="Filter supplier..."
        />
      )}
    </>
  );
}
```

File: src/components/supplier-table/SupplierTable.tsx [hunk 2]
```diff
@@ -90,157 +63,33 @@ export const columns: ColumnDef<Supplier>[] = [
...
-            table.getColumn("name")?.setFilterValue(event.target.value)
-          }
-          className="max-w-sm"
+    <>
+      {isLoading ? (
+        <Spinner />
+      ) : (
+        <TableComponent
+          columns={columns}
+          dataFetcher={() => Promise.resolve(tableData)}
```
Comment (bhausleitner) [inline src/components/supplier-table/SupplierTable.tsx:89]: Passing down a promise makes the TableComponent more complex than it needs to be, I think we can just pass it the actual data that we receive from api.part.partsBySupplier.useQuery();

You should be able to omit the the `useFetchParts` function replace the `SupplierPartTable` by something like this: 

```ts
export function SupplierTable() {
  const { data, isLoading } = api.supplier.getAllSuppliers.useQuery();
  return (
    <>
      {isLoading ? (
        <Spinner />
      ) : (
        <TableComponent
          columns={columns}
          data={data}
          filterPlaceholder="Filter supplier..."
        />
      )}
    </>
  );
}
```

File: src/components/supplier-detail/SupplierPartTable.tsx (hunk 1 above)
Comment (bhausleitner) [inline src/components/supplier-detail/SupplierPartTable.tsx:62]: same here

File: src/components/common/TableComponent.tsx [hunk 3]
```diff
@@ -25,58 +21,47 @@ import {
...
-  const { data, isLoading } = api.part.partsBySupplier.useQuery();
+  const [tableData, setTableData] = useState<T[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
 
   useEffect(() => {
-    if (data) {
-      setTableData(data);
-    }
-  }, [data]);
+    const fetchData = async () => {
```
Comment (bhausleitner) [inline src/components/common/TableComponent.tsx:52]: We can get rid of this to safe on re-rendering

File: src/pages/suppliers/[supplierId].tsx [hunk 4]
```diff
@@ -20,7 +20,7 @@ const SupplierPage = () => {
   });
 
   if (isLoading) {
-    return <p>Loading...</p>;
+    return <p></p>;
```
Comment (bhausleitner) [inline src/pages/suppliers/[supplierId].tsx:23]: Let's show the `<Spinner />` component here instead.

File: src/components/common/TableLink.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { useRouter } from "next/router";
+
+interface LinkProps {
+  href: string;
+  text: string;
+}
+
+const TableLink: React.FC<LinkProps> = ({ href, text }) => {
```
Comment (bhausleitner) [inline src/components/common/TableLink.tsx:9]: We haven't used the `React.FC<...>` generic in the code base yet. I personally would refrain from using it out of consistency reason. 

You can do the same thing with: 

```
const TableLink = ({ href, text }: LinkProps) => {
...
```

File: src/pages/suppliers/[supplierId].tsx [hunk 6]
```diff
@@ -20,7 +21,7 @@ const SupplierPage = () => {
   });
 
   if (isLoading) {
-    return <p>Loading...</p>;
+    return <Spinner></Spinner>;
```
Comment (bhausleitner) [inline src/pages/suppliers/[supplierId].tsx:24]: nit: Better to do `<Spinner />`

File: src/components/supplier-detail/SupplierTabs.tsx [hunk 7]
```diff
@@ -12,7 +12,7 @@ export function SupplierTabs() {
           <TabsTrigger value="quotes">Quotes</TabsTrigger>
         </TabsList>
         <TabsContent value="parts" className="space-y-4">
-          <SupplierParts></SupplierParts>
+          <SupplierPartTable></SupplierPartTable>
```
Comment (bhausleitner) [inline src/components/supplier-detail/SupplierTabs.tsx:15]: Same here, better to do `<SupplierPartTable />`

Comment (rabanspiegel) [reply src/components/supplier-detail/SupplierTabs.tsx:15]: I see! Will implement!

**Review by bhausleitner — APPROVED** [review]

**Pull request merged**